ipython/fragment_reattachment_example.ipynb

```diff
@@ -0,0 +1,1102 @@
+      "WARNING:root:No normalization for BCUT2D_CHGHI\n",
+      "WARNING:root:No normalization for BCUT2D_CHGLO\n",
+      "WARNING:root:No normalization for BCUT2D_LOGPHI\n",
+      "WARNING:root:No normalization for BCUT2D_LOGPLOW\n",
+      "WARNING:root:No normalization for BCUT2D_MRHI\n",
+      "WARNING:root:No normalization for BCUT2D_MRLOW\n",
+      "WARNING:root:No normalization for AvgIpc\n"
+     ]
+    }
+   ],
+   "source": [
+    "from rdkit.Chem.rdmolops import *\n",
+    "from rmgpy.molecule.fragment import Fragment\n",
+    "from rmgpy.tools.canteramodel import Cantera\n",
+    "from rmgpy.chemkin import *\n",
+    "import re\n",
+    "import os\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import sys\n",
+    "rmgpy_loc = \"/home/gridsan/adoner/RMG-Py/\"\n",
+    "sys.path.append(rmgpy_loc)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "045d6ebe-c055-42bc-ae55-643599631ea3",
+   "metadata": {},
+   "source": [
+    "## Defining the class FragList\n",
+    "The `FragList` object stores molecule and fragment concentrations from a single time step and provides a series of functions that groups fragments to form molecules. The procedure is described later on in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "9802cd48-4a7b-4244-acf8-cba0fc1aab69",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class FragList():\n",
+    "    '''\n",
+    "    to instantiate a FragList:\n",
+    "    fl = Fraglist(frag_list) \n",
+    "    where frag_list is a list of tuples of fragments and their amounts\n",
+    "    '''\n",
+    "\n",
+    "    def __init__(self, frag_list):\n",
+    "        self.raw = frag_list\n",
+    "\n",
+    "    def sort(self):\n",
+    "        '''\n",
+    "        sort a FragList into\n",
+    "            general_R_list - 2R fragments\n",
+    "            general_L_list - 2L fragments\n",
+    "            rr_ll_list - 2R or 2L fragments\n",
+    "            r_l_moles - 1R and 1L fragments\n",
+    "            multi_label_frag_3 - fragments with 3 cutting labels \n",
+    "            multi_label_frag_4 - fragments with 4 cutting labels\n",
+    "\n",
+    "            note: in our experience fragments with more than 4 cutting labels has never happened, but a warning will be printed if it does happen\n",
+    "        '''\n",
+    "        moles_remain = []\n",
+    "        one_R_dict = {}\n",
+    "        one_L_dict = {}\n",
+    "        general_R_list = []\n",
+    "        general_L_list = []\n",
+    "        rr_ll_list = []\n",
+    "        r_l_moles = []\n",
+    "        multi_label_frag_3 = []\n",
+    "        multi_label_frag_4 = []\n",
+    "\n",
+    "        for i, item in enumerate(self.raw):\n",
+    "            frag, amt = item\n",
+    "            if amt > 1e-6 and '[' not in frag:\n",
+    "                if len(re.findall(r'R', frag)) == 0 and len(re.findall(r'L', frag)) == 0:\n",
```

Comment:
Rather than repeatedly doing `re.findall` (and more importantly, for code clarity), it would be better to assign the output of this expression to a well-named variable at the beginning of each iteration of the loop and then re-use it, i.e.:
```
count_of_L_labels = re.findall(r'L', frag)
count_of_R_labels = re.findall(r'R', frag)
...
if count_of_R_labels == 0 and count_of_L_labels == 0:
...
```